provider/pipeline_stage_resource.go

```diff
@@ -77,8 +77,14 @@ func resourcePipelineStageRead(d *schema.ResourceData, m interface{}, createStag
 
 	var cStage client.Stage
 	cStage, err = pipeline.GetStage(d.Id())
+	if err == client.ErrStageNotFound {
+		log.Println("[WARN] No Pipeline Stage found")
+		d.SetId("")
```

Comment:
is it important to set empty Id ?

Reply:
Yes. The ID is the information used to identify if resource exist or not.
https://www.terraform.io/docs/extend/writing-custom-providers.html#error-handling-amp-partial-state